File: python_samples/list_books.py
#!/usr/bin/python
from sys import argv
from os import path

# Uncomment next two lines to disable warnings for insecure connections
# from urllib3 import disable_warnings, exceptions as urllib3_exceptions
# disable_warnings(urllib3_exceptions.InsecureRequestWarning)


from requests import get, post, exceptions
import json

from config import FQDN, PROTOCOL, PORT, USERNAME, PASSWORD, get_url, get_token

def get_documents(token, patient_id):
    return get(
            get_url('documents/list?PID=' + patient_id),
            verify = False,
            headers = { 'Authorization': 'Bearer ' + token }
            ).json()

def main():
    if len(argv) != 2:
        print(' ERROR - Usage: ' + path.basename(argv[0]) + ' <patient_id>')
        exit(0)

    token = get_token()
    patient_id = argv[1]
    result = get_documents(token, patient_id)
    for item in result['results'][1]['DocumentList']:
        doc_id = item['IDDOC']
        description = item['StudyDescription']
        doc_size = item['DocumentSize']
        print('ID: ' + doc_id, ' - Size: ' + doc_size, ' - Description: ' + description)

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print('Command ended with keyboard interrupt')
        try:
            exit(0)
        except SystemExit:
            _exit(0)

File: php_examples/list_books.php
<?php

require __DIR__ .'/config.php';

function getDocuments($token, $patientId)
{
    $response = httpGet(getUrl('documents/list?PID=' . $patientId), $token);
    // var_dump($response->results[1]);
    return $response->results[1]->DocumentList;
}

$documents = getDocuments(getToken(), '1234567890');
foreach ($documents as $item) {
    echo('ID:' . $item->IDDOC . " - Size: " . $item->DocumentSize . " - Description: " . $item->StudyDescription . "\n");
}
?>
